script.service.hue/resources/lib/reporting.py

```diff
@@ -0,0 +1,43 @@
+import json
+import platform
+import sys
+
+import xbmc
+import xbmcgui
+
+import rollbar
+from . import logger, ADDON, ADDONVERSION, ROLLBAR_API_KEY, ADDONID,KODIVERSION, ADDONPATH
+
+def _kodi_version():
+    query = dict(jsonrpc='2.0',
+                 method='Application.GetProperties',
+                 params=dict(properties=['version', 'name']),
+                 id=1)
+    response = json.loads(xbmc.executeJSONRPC(json.dumps(query)))
+    return response['result']['version']
+
+
+def error_report_requested(exc):
+    return xbmcgui.Dialog().yesno(
+        heading="{} {}".format(ADDONID, ADDON.getLocalizedString(30043)),
+        message=ADDON.getLocalizedString(30080) +
+        "\n[COLOR=red]{}[/COLOR]\n".format(exc) +
+        ADDON.getLocalizedString(30081)
+    )
+
+
+def report_error(url=None):
+    data = {
+        'machine': platform.machine(),
+        'platform': platform.system(),
+        'kodi': KODIVERSION,
+        #'kodi': _kodi_version(),
+        'url': url,
+    }
+    rollbar.init(ROLLBAR_API_KEY, capture_ip="anonymize", code_version=ADDONVERSION, root=ADDONPATH, scrub_fields='bridgeUser')
+    rollbar.report_exc_info(sys.exc_info(), extra_data=data)
+
```

Comment:
this really needs to be opt-in!

i would suggest to add a setting which a user can enable if they wish to submit error data.

Reply:
When there's an unhandled exception, an error message with the exception summary offers both 'cancel' and 'send report'. I preferred to ask every time rather than have an option people might forget. The Rollbar API is only initialized after they choose to send the report. 

The logic for this is in reporting.py, the dialogue box is on line 21 and check is on line 42.

Reply:
cool, thanx for the explanation. that way is certainly acceptable.